base/floatfuncs.jl

```diff
@@ -20,6 +20,7 @@ maxintfloat(::Type{Float64}) = 9007199254740992.
 maxintfloat(::Type{Float32}) = Float32(16777216.)
 maxintfloat(::Type{Float16}) = Float16(2048f0)
 maxintfloat(x::T) where {T<:AbstractFloat} = maxintfloat(T)
+maxintfloat(::Type{S}, ::Type{T}) where S <: AbstractFloat where T <: Integer = (m = maxintfloat(S); Int === Int32 ? min(m, S(typemax(T))) : m)
```

Comment:
It's good to know, thanks. Looking at it, I prefer the curly brace way because it's much easier to read; I'll default to it in future. I've changed the PR to that style.